CMakeLists.txt

```diff
@@ -172,9 +172,23 @@ endif()
 find_package(PythonInterp)
 if(PYTHONINTERP_FOUND)
   enable_testing()
-  add_test(NAME iwyu_tests
-    COMMAND ${PYTHON_EXECUTABLE} run_iwyu_tests.py -- $<TARGET_FILE:include-what-you-use>
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+
+  function(ADD_IWYU_TEST NAME FILE)
+    add_test(NAME ${NAME}
+      COMMAND ${PYTHON_EXECUTABLE} run_iwyu_tests.py --run-test-file=${FILE} -- $<TARGET_FILE:include-what-you-use>
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+  endfunction()
+
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} run_iwyu_tests.py --list-test-files
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE TEST_NAMES_AND_FILES)
+  string(REPLACE "\n" ";" TEST_NAMES_LIST ${TEST_NAMES_AND_FILES})
+  foreach (TEST_NAME_AND_FILE ${TEST_NAMES_LIST})
```

Comment:
```suggestion
  foreach (TEST_NAME_AND_FILE IN LISTS TEST_NAMES_LIST)
```

Reply:
It's not clear to me why this is better? The `IN LISTS` syntax mostly seems to be intended to deal with nested lists, and `TEST_NAMES_LIST` is just a plain list, isn't it?

Reply:
It just gives more hints about semantic to the reader... The older syntax considered if not deprecated but at least is not a _Modern CMake_ approach. For ~plain lists~ literal items there is `foreach(... IN ITEMS ...)`.